packages/contracts/src/strategies/BaseStrategy.sol

```diff
@@ -335,6 +338,18 @@ abstract contract BaseStrategy is
 
     /// @dev Fallback function that is called when the health check fails.
     function healthCheckFailedFallback() internal virtual override{
-        shutdown();
+        if (!paused()) {
+            shutdown();
+            _work();
+        }
+    }
+
+    /// @notice Returns free balance of the strategy.
+    function _underlyingBalance()
```

Comment:
Let's use `_freeAssets()` simular to what we have for the Vault
https://github.com/eonian-core/farm/blob/staging/packages/contracts/src/Vault.sol#L261-L263

Reply:
And also need simple test for this method